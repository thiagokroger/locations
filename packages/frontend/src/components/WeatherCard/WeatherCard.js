import React from 'react'
import PropTypes from 'prop-types'
import getCurrentTimeWithUTC from '../../utils/getCurrentTimeWithUTC'

const WeatherCard = ({ clime, observationTime, city, temperature, removeSearchValue }) => {
  return (
    <div className="card">
      <div className="row align-right">
        <span className="clickable" onClick={() => removeSearchValue(city)}><i className="fas fa-times" /></span>
      </div>
      <div className="row">
        {clime.map((value, i) => <span key={i}>{value}</span>)}
        <span>{city}</span>
      </div>
      <div className="row">
        <span className="temperature">{temperature}Â°C</span>
        <span>{getCurrentTimeWithUTC(parseInt(observationTime))}</span>
      </div>
    </div>
  )
}

WeatherCard.propTypes = {
  clime: PropTypes.array,
  observationTime: PropTypes.string,
  city: PropTypes.string,
  temperature: PropTypes.number,
  removeSearchValue: PropTypes.func.isRequired
}

WeatherCard.defaultProps = {
  clime: [],
  observationTime: '',
  city: '',
  temperature: 0
}

export default WeatherCard
