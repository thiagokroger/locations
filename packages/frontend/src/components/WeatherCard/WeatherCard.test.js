import React from 'react'
import { shallow } from 'enzyme'
import WeatherCard from './WeatherCard'

jest.mock('../../utils/getCurrentTimeWithUTC.js', () => jest.fn())

describe('Components - Weather Card without not required props', () => {
  it('should render without crashing', () => {
    const component = shallow(
      <WeatherCard removeSearchValue={jest.fn()} />
    )
    expect(component).toMatchSnapshot()
  })
})

describe('Components - Weather Card with all props', () => {
  it('should render without crashing', () => {
    const component = shallow(
      <WeatherCard removeSearchValue={jest.fn()} clime={['Clime test']} city="City test" observationTime="1" temperature={25}/>
    )
    expect(component).toMatchSnapshot()
  })

  it('should trigger removeSearchValue correctly', () => {
    const mockFunc = jest.fn()

    const component = shallow(
      <WeatherCard removeSearchValue={mockFunc} clime={['Clime test']} city="City test" observationTime="1" temperature={25}/>
    )

    expect(component).toMatchSnapshot()
    component.find('span').at(0).simulate('click')
    expect(mockFunc.mock.calls.length).toBe(1)
  })
})
