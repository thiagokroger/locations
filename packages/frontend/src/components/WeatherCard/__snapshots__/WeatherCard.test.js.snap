// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Weather Card with all props should render without crashing 1`] = `
<div
  className="card"
>
  <div
    className="row align-right"
  >
    <span
      className="clickable"
      onClick={[Function]}
    >
      <i
        className="fas fa-times"
      />
    </span>
  </div>
  <div
    className="row"
  >
    <span
      key="0"
    >
      Clime test
    </span>
    <span>
      City test
    </span>
  </div>
  <div
    className="row"
  >
    <span
      className="temperature"
    >
      25
      °C
    </span>
    <span />
  </div>
</div>
`;

exports[`Components - Weather Card with all props should trigger removeSearchValue correctly 1`] = `
<div
  className="card"
>
  <div
    className="row align-right"
  >
    <span
      className="clickable"
      onClick={[Function]}
    >
      <i
        className="fas fa-times"
      />
    </span>
  </div>
  <div
    className="row"
  >
    <span
      key="0"
    >
      Clime test
    </span>
    <span>
      City test
    </span>
  </div>
  <div
    className="row"
  >
    <span
      className="temperature"
    >
      25
      °C
    </span>
    <span />
  </div>
</div>
`;

exports[`Components - Weather Card without not required props should render without crashing 1`] = `
<div
  className="card"
>
  <div
    className="row align-right"
  >
    <span
      className="clickable"
      onClick={[Function]}
    >
      <i
        className="fas fa-times"
      />
    </span>
  </div>
  <div
    className="row"
  >
    <span />
  </div>
  <div
    className="row"
  >
    <span
      className="temperature"
    >
      0
      °C
    </span>
    <span />
  </div>
</div>
`;
